### PR TITLE
Add issue events API

### PR DESCRIPTION
**Pull request by martinvanzijl**
Fix for issue #376 (add issue events API).

Added the following methods:

 - GHIssue.listEvents()
 - GHRepository.listIssueEvents()
 - GHRepository.getIssueEvent(int id)

### CODE REVIEW

File: src/main/java/org/kohsuke/github/GHRepository.java [hunk 1]
```diff
@@ -1597,4 +1597,89 @@ String getApiTailUrl(String tail) {
         if (tail.length()>0 && !tail.startsWith("/"))    tail='/'+tail;
         return "/repos/" + getOwnerName() + "/" + name +tail;
     }
+
+    public static class IssueEventInfo {
```
Comment (bitwiseman) [inline src/main/java/org/kohsuke/github/GHRepository.java]: @martinvanzijl 
Does this need to be a child class inside `GHRepository`?

Comment (martinvanzijl) [reply src/main/java/org/kohsuke/github/GHRepository.java]: @bitwiseman 
I've moved it to a separate class called `GHIssueEvent`.

**Comment by martinvanzijl** [comment]
I've uploaded my latest progress.

I want to make the unit tests more robust by creating an issue within the test itself, editing it and making sure the appropriate events are created. I also want to test the fields of the `GHIssueEvent` object itself.

**Comment by bitwiseman** [comment]
Cool just tell me when you're ready for review.

**Comment by martinvanzijl** [comment]
@bitwiseman 
I've updated the unit tests. Can you please review?

**Pull request merged**